dissect/target/tools/info.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import argparse
+import json
+import logging
+
+from dissect.target import Target
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.tools.query import record_output
+from dissect.target.tools.utils import (
+    configure_generic_arguments,
+    process_generic_arguments,
+)
+
+InfoRecord = TargetRecordDescriptor(
+    "target/info",
+    [
+        ("datetime", "last_activity"),
+        ("datetime", "install_date"),
+        ("net.ipaddress[]", "ips"),
+        ("string", "os_family"),
+        ("string", "os_version"),
+        ("string", "architecture"),
+        ("string", "language"),
+        ("string", "timezone"),
+    ],
+)
+
+
+log = logging.getLogger(__name__)
+logging.lastResort = None
+logging.raiseExceptions = False
+
+
+def main():
+    """target-info"""
+
+    help_formatter = argparse.ArgumentDefaultsHelpFormatter
+    parser = argparse.ArgumentParser(
+        description="dissect.info",
+        fromfile_prefix_chars="@",
+        formatter_class=help_formatter,
+        add_help=False,
+    )
+    parser.add_argument("targets", metavar="TARGETS", nargs="*", help="Targets to display info from")
+    parser.add_argument("--from-file", nargs="?", help="file containing targets to load")
+    parser.add_argument("-d", "--delimiter", default=" ", action="store", metavar="','")
+    parser.add_argument("-s", "--strings", action="store_true", help="print output as string")
+    parser.add_argument("-r", "--record", action="store_true", help="print output as record")
+    parser.add_argument("-j", "--json-pretty", action="store_true", help="output records as pretty json")
+    parser.add_argument("-J", "--json", action="store_true", help="output records as one-line json")
+    configure_generic_arguments(parser)
+
+    args, rest = parser.parse_known_args()
+
+    if not args.targets and ("-h" in rest or "--help" in rest):
+        parser.print_help()
+        parser.exit()
+
+    process_generic_arguments(args)
+
+    if not args.targets and not args.from_file:
+        parser.error("too few arguments")
+
+    if args.from_file:
+        with open(args.from_file, "r") as f:
+            targets = f.read().splitlines()
+            while targets[-1] == "":
+                targets = targets[:-1]
+            args.targets = targets
+
+    for target in Target.open_all(args.targets):
+
+        if args.json:
+            print(json.dumps(obj_target_info(target)))
+        elif args.json_pretty:
+            print(json.dumps(obj_target_info(target), indent=4))
+        elif args.record:
+            rs = record_output(args.strings, args.json)
+            rs.write(
+                InfoRecord(
+                    os_family=target.os,
+                    os_version=target.version,
+                    ips=list(set(target.ips + target.ips_dhcp)),
```

Comment:
`ips_dhcp` does not exist on Windows targets and thus will error.

You can use `target.has_function("ips_dhcp")` to check if the target has the supported function.

Reply:
For API consistency one could argue Windows targets should also have a `ips_dhcp` function. Currently `ips` for Windows targets will also return DHCP IPs. 

Do you prefer checking if `ips_dhcp` exists for the current target or adding `ips_dhcp` to `plugins/os/windows/_os.py`?

Reply:
I agree. Technically everything here that isn't defined in `OSPlugin` is not guaranteed to exist, but we can start formalizing most of those soon.

I'm very torn on `ips_dhcp`. Can you give an indication on the time it takes to parse a decently sized syslog? If it's not very long, maybe I'm going back on my initial decision to separate it out, eat the performance hit now, and properly fix it later with more elaborate network interface code.

Reply:
Did some small tests with uncompressed logs ranging from 1MB to 10MB. The time it takes to parse a syslog (in seconds) seems directly proportional to the log size (in MBs). Meaning that 1MB of syslog equals to roughly 1 second of parsing time in Dissect.

It's difficult to say how large on average syslogs are on Unix systems, as the size is heavily dependent on the specifics of what the machine is used for. Depending on configuration, `logrotated` is usually configured to rotate logs every day or week, regardless of size, and keep logs for four weeks.

Reply:
Had a short internal discussion, let's eat the performance hit now and fix it properly later with more elaborate network interface code. That means reverting the change in #133 and putting the DHCP logic back into the `ips` function.

Reply:
Implemented in [4584d9b](https://github.com/fox-it/dissect.target/pull/136/commits/4584d9bdad96a4b66c0c1dbcc142d0d0b4281199) and [2c3216e](https://github.com/fox-it/dissect.target/pull/133/commits/2c3216ea1aec73a55875698ec84162709ff0a994).